python/grass/temporal/core.py

```diff
@@ -759,22 +775,32 @@ def init(raise_fatal_error=False, skip_db_version_check=False):
                         }
                     )
                 )
-            if "tgis_db_version" in entry and entry[1] != str(get_tgis_db_version()):
+
+        # temporal database version check
+        tgis_db_version_meta = get_tgis_db_version_from_metadata(metadata)
+        if tgis_db_version_meta != tgis_db_version:
+            message = _(
+                "Temporal database version mismatch detected."
+                "\n {backup}Supported temporal database version"
+                " is: {tdb}\nCurrent temporal database info:"
```

Comment:
Done in 0838a7fdc. 

```
WARNING: Temporal database version mismatch detected.
         Please create a backup of your temporal database to avoid losing
         data.
         SOLUTION: Run t.upgrade command in order to upgrade your temporal
         database.
         Supported temporal database version is: 3
         Your existing temporal database version: 2
         Current temporal database info:
         DBMI interface:..... sqlite3
         Temporal database:..
         /home/martin/grassdata/tgis/PERMANENT/tgis/sqlite.db
```